src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -138,20 +140,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
+    String type = request.getParameter("type");
     String messageContent = request.getParameter("message");
 
-    // this removes any HTML from the message content
-    String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
+    if(type != null){
+      if(!conversation.getType().equals(type)){
+        conversation.makePrivate();
```

Comment:
So it's a one-directional setting? Are you planning to allow users to switch a conversation back to being public?

Reply:
I was planning on implementing that too. I will add it to this PR.